controllers/openshift_controller.go

```diff
@@ -428,6 +545,19 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	ds := r.processDaemonsetForMonitor()
+	if ds == nil {
+		r.Log.Error(err, "error deleting monitor Daemonset")
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 15}, nil
```

Comment:
Two semantic questions
1. Should the error message be more aligned to DS being nil ? May be something like "received empty Daemonset"?
2. Should this be requeue with error ?

Reply:
Wondering if this can happen (processDaemonsetForMonitor() returning nil) or is a check to make the code more robust for the future.